fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -90,12 +90,12 @@ def __init__(self,
         self.grid_offset = grid_offset
         self.grid = grid
 
-        self._best_grid_index = None
-        self._predictors = []
-        self._lambda_vecs = pd.DataFrame()
-        self._objectives = []
-        self._gammas = pd.DataFrame()
-        self._oracle_execution_times = []
+        self.best_grid_index_ = None
+        self.predictors_ = []
+        self.lambda_vecs_ = pd.DataFrame()
+        self.objectives_ = []
+        self.gammas_ = pd.DataFrame()
+        self.oracle_execution_times_ = []
```

Comment:
There's a lot of stuff in this `__init__` that should go into `fit`. But we can hold off for a future PR?

Reply:
Just did that locally. :-) Thanks!

Reply:
this is reasonable. there's also the validation code and a conversion using `float`, but we can tackle that in a future PR.

Reply:
The validation code needs to be addressed here because it requires initialized members which we don't have anymore now :-)